### PR TITLE
Add PV test to persistent volume integration test

### PR DESCRIPTION
**Pull request by rootfs**
This is a follow-up to #25120

@kubernetes/sig-storage

### CODE REVIEW

File: test/integration/persistent_volumes_test.go [hunk 1]
```diff
@@ -177,6 +178,87 @@ func TestPersistentVolumeRecycler(t *testing.T) {
...
+			t.Fatalf("Unexpected error getting pv: %v", err)
+		}
+		if pv.Spec.ClaimRef == nil {
+			continue
+		}
+		// found a bounded PV
+		if pv.Spec.ClaimRef.Name != pvc.Name {
+			t.Fatalf("Bind mismatch! Expected %s but got %s", pvc.Name, pv.Spec.ClaimRef.Name)
+		}
+		t.Logf("claim bounded to %s", pv.Name)
```
Comment (jsafrane) [inline test/integration/persistent_volumes_test.go]: You should check that the smallest matching volume is bound.

**Comment by jsafrane** [comment]
While I like idea of the new test, I do not like copy-paste style - we will add new tests here and we need a better way how to add new ones. What about `newVolume(size, name)`, `newPVC(size, name)`, `newController(kubeClient)` and so on?

File: test/integration/persistent_volumes_test.go [hunk 2]
```diff
@@ -177,6 +161,131 @@ func TestPersistentVolumeRecycler(t *testing.T) {
...
+		if pv.Spec.ClaimRef == nil {
+			continue
+		}
+		// found a bounded PV
+		p := pv.Spec.Capacity[api.ResourceStorage]
+		pvCap := p.Value()
+		p = resource.MustParse(strconv.Itoa(maxPVs/2) + "G")
+		pvcCap := p.Value()
+		if pv.Spec.ClaimRef.Name != pvc.Name || pvCap != pvcCap {
+			t.Fatalf("Bind mismatch! Expected %s capacity %v but got %s capacity %v", pvc.Name, resource.MustParse(strconv.Itoa(50)+"G"),
```
Comment (jsafrane) [inline test/integration/persistent_volumes_test.go]: This part is a bit sketchy, what about:

```
pvCap := pv.Spec.Capacity[api.ResourceStorage)
expectedCap := resource.MustParse(strconv.Itoa(maxPVs/2) + "G")
if pv.Spec.ClaimRef.Name != pvc.Name || pvCap.Value() != pvcCap.Value() {
			t.Fatalf("Bind mismatch! Expected %s capacity %v but got %s capacity %v", pvc.Name, expectedCap, pv.Spec.ClaimRef.Name, pvCap)
```

**Comment by jsafrane** [comment]
LGTM with one minor nit

**Comment by rootfs** [comment]
thanks, updated @jsafrane

**Comment by jsafrane** [comment]
LGTM
I'll follow up with additional tests shortly.

**Comment by jsafrane** [comment]
@k8s-bot test this issue: #25037

**Comment by jsafrane** [comment]
@k8s-bot test this issue: #25461

**Comment by jsafrane** [comment]
@k8s-bot test this issue #25652

**Comment by jsafrane** [comment]
@k8s-bot test this issue: #IGNORE

**Comment by jsafrane** [comment]
restoring lgtm

**Comment by pmorie** [comment]
@k8s-bot test this issue: #25879

**Comment by pmorie** [comment]
@rootfs integration tests failed in jenkins, PTAL

**Comment by rootfs** [comment]
It passed in my setup. Last message before error is the following. It happens to some other PRs too. 

```text
E0519 11:35:35.844161   18503 event.go:207] Unable to write event: 'dial tcp 127.0.0.1:43859: getsockopt: connection refused' (may retry after sleeping)
```

**Comment by jsafrane** [comment]
@k8s-bot test this issue: #IGNORE
this happened in #25263 several times too, does not seem to be related to this PR

**Pull request merged**